### ISSUE TITLE
You are running an unknown version of Mylar. Run the updater to identify your version.

### ISSUE THREAD

**Issue by royaljack** [evilhero/mylar#2120]
Good morning, Beginning a few days ago after updating to the newest version of git on windows 10 I've been running into an issue with updates no longer working. "You're running an unknown version of Mylar. Update or Close" keeps popping up. If I click Update the pages goes to the waiting 60 seconds to restart message as per normal but the update is failing. I have deleted my entire mylar directory and did a fresh clone from github (for testing did not use my old config, cache, or db). Booted up mylar in the default configuration and I get the same unknown version of Mylar messaging but for Master instead of dev. I then tried to copy over my cache, config, and db from my old install and I'm back at the beginning. I'm wondering if the newest version of Git for Windows 10 is causing the issue? I did also verify my GIT path's in windows environment as well as that it was hard coded in my config file correctly. Any ideas?

24-Nov-2018 10:24:18 - ERROR :: mylar.getVersion.81 : MAIN : Couldn't find latest installed version.
24-Nov-2018 10:24:18 - ERROR :: mylar.getVersion.93 : MAIN : Output does not look like a hash, not using it
24-Nov-2018 10:24:18 - INFO :: mylar.versionload.309 : MAIN : Version information: development [None]
24-Nov-2018 10:24:19 - INFO :: mylar.checkGithub.201 : MAIN : You are running an unknown version of Mylar. Run the updater to identify your version

**Comment by royaljack** [comment]
Oh also when I perform the update command I often now see this error mixed in with the unknown version messages. 

2018-11-24 10:35:19 | ERROR | Output does not look like a hash, not using it
-- | -- | --
2018-11-24 10:35:19 | ERROR | Couldn't find latest installed version.

**Comment by evilhero** [comment]
As long as your git_branch in your config.ini is pointing to the correct branch that you cloned (ie. ```git_branch = development```), it should be able to figure things out on it's own. When Mylar performs an update check it uses the git_branch value to do the check, so if you have it set to Master (or None by default I believe) and you cloned the development branch, it will always say you're out of date. If the value is None, performing the update will fix the branch information within Mylar so it shows the value in the config.ini.

That being said, I do remember something about git removing some SSL ciphers that were no longer needed, and it caused some issues - but I cannot remember if it was only for *nix based clients.

**Comment by royaljack** [comment]
It sounds like I did try to clone again from development and start from
scratch.

**Comment by evilhero** [comment]
Can you start Mylar with the -v flag (or if you already have it running, just go to the Logs tab and toggle the verbose logging on) and then try to do the update again. There might be more info behind the scenes happening that's causing it to fail.

Do you run Mylar using the git shell option within Git for Windows or do you just do it via the cmd prompt ?

Can you paste the [git] section from your config.ini  ?

**Comment by pipieye** [comment]
Issue happened on my last update as well. Here is the log with the -v option that has the error.
Running mylar on MAC using terminal to start via the command "Python -v mylar.py"
Nothing modified on the config.ini that was running fine previously, even updates.

# /usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/lib/python2.7/stringprep.pyc matches /usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/lib/python2.7/stringprep.py
import stringprep # precompiled from /usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/lib/python2.7/stringprep.pyc
26-Nov-2018 16:39:15 - ERROR :: mylar.getVersion.93 : MAIN : Output does not look like a hash, not using it
26-Nov-2018 16:39:15 - INFO :: mylar.versionload.309 : MAIN : Version information: development [None]
# /usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/lib/python2.7/netrc.pyc matches /usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/lib/python2.7/netrc.py
import netrc # precompiled from /usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/lib/python2.7/netrc.pyc
26-Nov-2018 16:39:15 - INFO :: mylar.checkGithub.201 : MAIN : You are running an unknown version of Mylar. Run the updater to identify your version
26-Nov-2018 16:39:15 - INFO :: mylar.start.386 : MAIN : [SEARCH-QUEUE] Attempting to background load the search queue....
26-Nov-2018 16:39:15 - INFO :: mylar.start.404 : MAIN : [WEEKLY] Checking for existance of Weekly Comic listing...
26-Nov-2018 16:39:15 - INFO :: mylar.start.434 : MAIN : [RSS-FEEDS] Initiating startup-RSS feed checks.
26-Nov-2018 16:39:15 - INFO :: mylar.start.437 : MAIN : [RSS-FEEDS] RSS last run @ 2018-11-26 16:31:07.966004
26-Nov-2018 16:39:15 - INFO :: mylar.start.464 : MAIN : Firing up the Background Schedulers now....
26-Nov-2018 16:39:15 - INFO :: mylar.start.468 : MAIN : Background Schedulers successfully started...

**Comment by royaljack** [comment]
This is what happens when I try to perform an update while mylar is already running. Like pipieye I've been running the development clone for months and doing regular updates without issue when prompted. The issue seemed to begin after updating the version of GIT on my windows 10 box a week or so ago.
2018-12-01 21:17:02 | INFO | [PULL-LIST] Populating & Loading pull-list data directly from alternate website
-- | -- | --
2018-12-01 21:17:02 | INFO | Weekly pull list present - checking if it's up-to-date..
2018-12-01 21:17:02 | INFO | [WEEKLY] Checking Weekly Pull-list for new releases/updates
2018-12-01 21:17:02 | INFO | Background Schedulers successfully started...
2018-12-01 21:17:02 | INFO | Firing up the Background Schedulers now....
2018-12-01 21:17:02 | INFO | [FOLDER MONITOR] Enabling folder monitor for : B:\NZBGet\complete\Mylar every 5 minutes.
2018-12-01 21:17:02 | INFO | [WEEKLY] Weekly Pull-Update initializing immediately as it has been 21.8557122222 hours since the last run
2018-12-01 21:17:02 | INFO | [WEEKLY] Checking for existance of Weekly Comic listing...
2018-12-01 21:17:02 | INFO | [SEARCH-QUEUE] Attempting to background load the search queue....
2018-12-01 21:17:02 | INFO | You are running an unknown version of Mylar. Run the updater to identify your version
2018-12-01 21:17:01 | INFO | Version information: development [None]
2018-12-01 21:17:01 | ERROR | Output does not look like a hash, not using it
2018-12-01 21:17:01 | ERROR | Couldn't find latest installed version.
2018-12-01 21:16:59 | INFO | Starting Mylar on http://0.0.0.0:8090/
2018-12-01 21:16:59 | INFO | [DIRECTORY-CHECK] Found comic directory: B:\mylar-dev\mylar
2018-12-01 21:16:59 | INFO | Sucessfully ordered 723 series in your watchlist.
2018-12-01 21:16:59 | INFO | Remapping the sorting to allow for new additions.
2018-12-01 21:16:59 | INFO | Successfully discovered local IP and locking it in as : 192.168.40.111
2018-12-01 21:16:59 | INFO | Correcting Null entries that make the main page break on startup.
2018-12-01 21:16:59 | INFO | Ensuring DB integrity - Removing all Erroneous Comics (ie. named None)
2018-12-01 21:16:58 | INFO | Populating Custom Exception listings into Mylar....
2018-12-01 21:16:58 | INFO | Populating Base Exception listings into Mylar....
2018-12-01 21:16:58 | INFO | Checking to see if the database has all tables....

**Comment by evilhero** [comment]
Try doing a command from the commandime within the root of the mylar directory:
```git rev-parse HEAD```

Ideally it should return the hash of the commit you're currently running. What I'm expecting is that it will provide some error as to what's happening.

**Comment by royaljack** [comment]
3e12599502bc99343da8564762bd46277f5cdd8c

**Comment by evilhero** [comment]
3e12599 is from the master branch, not the development.

Quite possibly, either a ```git checkout development``` wasn't issued after the initial clone, and/or the git_branch in the config.ini isn't set to the correct branch that the git clone is currently set to.

If you do a ```git status ¦more``` from the cli, you can see what branch you're currently on.

You can also try doing both a ```git fetch``` followed by a ```git pull``` in order to update to the latest. If there's a problem with your git installation, it will occur at this point most likely. 

However I believe being on the Master branch while your git_branch is set to Development is probably the more likely culprit (because it's checking the development branch for your current commit and can't locate it - as it's on the master branch)

**Comment by pipieye** [comment]
In my case, looks like my xcode went haywire when I tried to upgrade xcode to higher version. So git is reporting cannot find commandlinetools. Fixing xcode first and see how it goes.

**Comment by pipieye** [comment]
installing the xcode command line tools fixed my issue. Thanks.

**Comment by royaljack** [comment]
Sorry to revive a dead thread. I got tired of fighting with the error and am just revisiting it. 

1. I completely deleted my entire mylar directory in windows 10 and emptied the recycle bin.
2. I uninstalled GIT.
3. Restarted PC, Reinstalled GIT.
4. I launch command prompt as admin
5. CD to B:\
6. execute "git clone https://github.com/evilhero/mylar.git" which does clones the repository into B:\Mylar
7. cd B:\mylar
8. "git checkout development" returns the following message "Switched to a new branch 'development'
Branch 'development' set up to track remote branch 'development' from 'origin'."
9. "python mylar.py -v" launches and mylar can now see its version!

In summary I believe that the reinstall of GIT is what resolved the issue as I had tried every other step multiple times!
